cmake/modules/platform/toolcfg/openxl.cmake

```diff
@@ -0,0 +1,145 @@
+# and is available at https://www.apache.org/licenses/LICENSE-2.0.
+#
+# This Source Code may also be made available under the following Secondary
+# Licenses when the conditions for such availability set forth in the
+# Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+# version 2 with the GNU Classpath Exception [1] and GNU General Public
+# License, version 2 with the OpenJDK Assembly Exception [2].
+#
+# [1] https://www.gnu.org/software/classpath/license.html
+# [2] https://openjdk.org/legal/assembly-exception.html
+#
+# SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
+###############################################################################
+
+
+if(OMR_OS_ZOS)
+	set(OMR_ZOS_COMPILE_ARCHITECTURE "arch10" CACHE STRING "z/OS compile machine architecture" FORCE)
+	set(OMR_ZOS_COMPILE_TARGET "ZOSV2R4" CACHE STRING "z/OS compile target operating system" FORCE)
+	set(OMR_ZOS_COMPILE_TUNE "12" CACHE STRING "z/OS compile machine architecture tuning" FORCE)
+	set(OMR_ZOS_LINK_COMPAT "ZOSV2R4" CACHE STRING "z/OS link compatible operating system" FORCE)
+	set(CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS "--shared")
+	set(CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS "--shared")
+
+	set(CMAKE_ASM_FLAGS "-fno-integrated-as")
+	string(APPEND CMAKE_ASM_FLAGS " \"-Wa,-mgoff\"")
+	if(OMR_ENV_DATA64)
+		string(APPEND CMAKE_ASM_FLAGS " \"-Wa,-mSYSPARM(BIT64)\"")
+	endif()
+
+	list(APPEND OMR_PLATFORM_COMPILE_OPTIONS
+		"-fstrict-aliasing"
+		"-mzos-target=${OMR_ZOS_COMPILE_TARGET}"
+		"-m64"
+	)
+
+	list(APPEND OMR_PLATFORM_C_COMPILE_OPTIONS
+		-march=${OMR_ZOS_COMPILE_ARCHITECTURE}
+	)
+
+	list(APPEND OMR_PLATFORM_CXX_COMPILE_OPTIONS
+		-march=${OMR_ZOS_COMPILE_ARCHITECTURE}
+		"-std=gnu++14"
+		-fasm
+	)
+
+	list(APPEND OMR_PLATFORM_SHARED_COMPILE_OPTIONS
+		-fvisibility=default
+	)
+
+	if(OMR_ENV_DATA64)
+		list(APPEND OMR_PLATFORM_DEFINITIONS
+			-DJ9ZOS39064
+		)
+	else()
+		list(APPEND OMR_PLATFORM_DEFINITIONS
+			-D_LARGE_FILES
+		)
+	endif()
+
+	# Testarossa build variables. Longer term the distinction between TR and the rest
+	# of the OMR code should be heavily reduced. In the mean time, we keep
+	# the distinction.
+
+	# TR_COMPILE_OPTIONS are variables appended to CMAKE_{C,CXX}_FLAGS, and so
+	# apply to both C and C++ compilations.
+	list(APPEND TR_COMPILE_OPTIONS
+		-DYYLMAX=1000
+	)
+
+	list(APPEND TR_CXX_COMPILE_OPTIONS
+		-Wuninitialized
+		-mnocsect
+	)
+
+	# Configure the platform dependent library for multithreading.
+	set(OMR_PLATFORM_THREAD_LIBRARY "")
+endif()
+
+set(SPP_CMD ${CMAKE_C_COMPILER})
+
+if(CMAKE_C_COMPILER_IS_XLCLANG)
+	# The -P option doesn't sit well with XLClang, so it's not included. It causes:
+	# "ld: 0711-317 ERROR: Undefined symbol: <SYMBOL>" when libj9jit29.so is getting linked.
+	set(SPP_FLAGS -E)
+else()
+	set(SPP_FLAGS -E -P)
+endif()
```

Comment:
Isn't `CMAKE_C_COMPILER_IS_XLCLANG` only true when `OMR_TOOLCONFIG` is `xlc` (in which case this file would not be used)?

Reply:
yes, will remove the conditional and keep the path thats always taken for open-xl.